peers.py

```diff
@@ -0,0 +1,58 @@
+from charms.reactive import RelationBase
+from charms.reactive import hook
+from charms.reactive import scopes
+from charms.reactive import not_unless
+
+
+class EtcdPeer(RelationBase):
+    ''' ETCD Peering works in tandem with the leader node to establish a cluster
+        as sanely as possible. This works statically behind a firewall without
+        any need for an external service to function as the discovery mechanism
+
+        Every peer participating in the cluster declares themselves to the
+        cluster. The leader then ingests the data and leader-set's the cluster
+        state (string, status, token) for the peers to ingest and participate.
+    '''
+    scope = scopes.UNIT
+
+    auto_accessors = ['public_address', 'unit_name', 'port']
+
+    @hook('{peers:etcd}-relation-joined')
+    def peers_joined(self):
+        # As i understand this, this only operates on the conversation in
+        # current scope.
+        conv = self.conversation()
+        conv.set_state('{relation_name}.declare_self')
+
+    @hook('{peers:etcd}-relation-changed')
+    def peers_changed(self):
+        # As i understand this, this only operates on the conversation in
+        # current scope.
+        conv = self.conversation()
+        # if we get a port, we'll assume we have all the data....
```

Comment:
The code is not checking for port here anymore. This comment does not match the code. That is worst kind of comment.